eclair-core/src/main/scala/fr/acinq/eclair/io/Switchboard.scala

```diff
@@ -55,33 +55,34 @@ class Switchboard(nodeParams: NodeParams, peerFactory: Switchboard.PeerFactory)
     peerChannels.keySet
   }
 
-  def receive: Receive = normal(initialPeersWithChannels)
+  def receive: Receive = normal(initialPeersWithChannels())
 
   def normal(peersWithChannels: Set[PublicKey]): Receive = {
 
     case Peer.Connect(publicKey, _, _) if publicKey == nodeParams.nodeId =>
       sender() ! Status.Failure(new RuntimeException("cannot open connection with oneself"))
 
     case Peer.Connect(nodeId, address_opt, replyTo) =>
-      // we create a peer if it doesn't exist
+      // we create a peer if it doesn't exist: when the peer doesn't exist, we can be sure that we don't have channels,
+      // otherwise the peer would have been created during the initialization step.
       val peer = createOrGetPeer(nodeId, offlineChannels = Set.empty)
-      val c = if (replyTo == ActorRef.noSender){
+      val c = if (replyTo == ActorRef.noSender) {
         Peer.Connect(nodeId, address_opt, sender())
-      }else{
+      } else {
         Peer.Connect(nodeId, address_opt, replyTo)
       }
       peer forward c
 
     case d: Peer.Disconnect =>
       getPeer(d.nodeId) match {
         case Some(peer) => peer forward d
-        case None => sender() ! Status.Failure(new RuntimeException("peer not found"))
+        case None => sender() ! Status.Failure(new RuntimeException(s"peer ${d.nodeId} not found"))
```

Comment:
I wanted to just return a `PeerNotFound` here, but it forces us to change the API code in `Eclair.scala`...
So I kept it for now, I think that's something we will need to do once we move this actor to akka typed.